c10/core/TensorImpl.h

```diff
@@ -1577,6 +1594,20 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return data_type_.itemsize();
   }
 
+  void set_backend_meta(intrusive_ptr<c10::BackendMeta> backend_meta) {
+    if (!extra_meta_) {
+      extra_meta_ = std::make_unique<ExtraMeta>();
+    }
+    extra_meta_->backend_meta_ = std::move(backend_meta);
+  }
+
+  intrusive_ptr<c10::BackendMeta> get_backend_meta() {
+    if (!extra_meta_) {
+      return nullptr;
+    }
+    return extra_meta_->backend_meta_;
```

Comment:
You probably want some way to get at BackendMeta without forcing a refcount bump

Reply:
True. Should I call it c10::BackendMeta* get_raw_backend_meta()?

Reply:
Unless you have a pressing need for it, I would just not offer the intrusive ptr accessor (and use the good name for the reference)

Reply:
We sometimes share this object between more than one TensorImpl, so keeping the get_backend_meta_intrusive_ptr